docs/guides/templates.md

```diff
@@ -0,0 +1,41 @@
+# Templates
+
+When a Function is created, an example implementation and a Function metadata file are written into the new Function's working directory.  Together, these files are referred to as the Function's Template.  Included are the templates 'http' and 'events' for each supported language runtime.
+
+These embedded templates are minimal by design.  The Function contains a minimum of external dependencies, and the 'func.yaml' defines a final environment within which the Funciton will execute that is devoid of any extraneous packages or services.
+
+To make use of more complex inital Function implementions, or to define runtime environments with arbitrarily complex requirements, the templates system is fully pluggable.
+
+## External Git Repositories
+
+When creating a new Function, a Git repository can be specified as the source for the template files.  For example, the Boson Project maintains a set of example Functions at https://github.com/boson-project/templates which can be used during project creation.
```

Comment:
Do we? I'd prefer to state this as forward looking, or remove it altogether.

Reply:
I just created that repository, with a "hello-world" template for each language as a placeholder.  Realistically it's going to be some time before there's anything actually there, but this paragraph does check out.

Would you prefer we omit this until such time as there's at least something _in_ the specified repo?   Pro: it illustrates the point.  Con: the illustration is an invisible sculpture.

Reply:
I think leaving it in is fine for now if it checks out. But it will probably change whenever we figure out what the real upstream situation is.